pkg/operation/cloudbotanist/alicloudbotanist/controlplane.go

```diff
@@ -165,6 +145,40 @@ func (b *AlicloudBotanist) GenerateKubeAPIServerExposeConfig() (map[string]inter
 	}, nil
 }
 
+// GenerateCSIConfig generates the configuration for CSI charts
+func (b *AlicloudBotanist) GenerateCSIConfig() (map[string]interface{}, error) {
+	conf := map[string]interface{}{
+		"regionID": b.Shoot.Info.Spec.Cloud.Region,
+		"credential": map[string]interface{}{
+			"accessKeyID":     base64.StdEncoding.EncodeToString(b.Shoot.Secret.Data[AccessKeyID]),
+			"accessKeySecret": base64.StdEncoding.EncodeToString(b.Shoot.Secret.Data[AccessKeySecret]),
+		},
+		"podAnnotations": map[string]interface{}{
+			"csiAttacher": map[string]interface{}{
+				fmt.Sprintf("checksum/%s", common.CSIAttacher): b.CheckSums[common.CSIAttacher],
+			},
+			"csiPluginAlicloud": map[string]interface{}{
+				fmt.Sprintf("checksum/%s", common.CloudProviderSecretName): b.CheckSums[common.CloudProviderSecretName],
+			},
+			"csiProvisioner": map[string]interface{}{
+				fmt.Sprintf("checksum/%s", common.CSIProvisioner): b.CheckSums[common.CSIProvisioner],
+			},
+			"csiSnapshotter": map[string]interface{}{
+				fmt.Sprintf("checksum/%s", common.CSISnapshotter): b.CheckSums[common.CSISnapshotter],
+			},
+		},
+		"enabled": true,
+	}
+
+	return b.ImageVector.InjectImages(conf, b.ShootVersion(), b.ShootVersion(),
+		common.CSIAttacherImageName,
+		common.CSIPluginAlicloudImageName,
```

Comment:
This is not needed here, right?

Reply:
We still need it. This function is used for 2 charts.